server/vulnerabilities/nvd/tools/cvefeed/cvecache.go

```diff
@@ -155,7 +155,14 @@ func (c *Cache) Get(cpes []*wfn.Attributes) []MatchResult {
 		c.mu.Unlock()
 		<-cves.ready
 		c.mu.Lock() // TODO: XXX: ugly, consider using atomic.Value instead
-		cves.evictionIndex = c.evictionQ.touch(cves.evictionIndex)
+		// Check if the item still exists in cache (it may have been evicted)
+		if _, exists := c.data[key]; exists {
```

Comment:
This looks like we have two rather different code paths depending on whether the data was in cache and got evicted and needs to be added back, or whether the data wasn't in the cache to begin with, and the latter has a bunch of extra steps that seem like they may be applicable.

Assigning this over to @ksykulev to pick through more carefully since if this fix introduces a regression it's going to be even more tricky to troubleshoot than the occasional panics.